Framework/DataHandling/src/SetSample.cpp

```diff
@@ -711,8 +712,10 @@ void SetSample::setSampleShape(API::ExperimentInfo &experiment, const Kernel::Pr
   // Try known shapes or CSG first if supplied
   if (args) {
     const auto refFrame = experiment.getInstrument()->getReferenceFrame();
-    const auto xml = tryCreateXMLFromArgsOnly(*args, *refFrame);
+    auto xml = tryCreateXMLFromArgsOnly(*args, *refFrame);
     if (!xml.empty()) {
+      std::vector<double> rotationMatrix = experiment.run().getGoniometer().getR();
```

Comment:
I think we should try and identify here whether the workspace has a goniometer set up with a non-default rotation matrix. It might have been helpful if getGoniometer returned nothing if this was the case, however it seems to be the case that the Run class create a default Goniometer object in its constructor.
Another way might be to see if the rotationMatrix returned is equal to an "identity" matrix or null rotation. We could then only insert the goniometer tag if it's a non-null rotation. This makes the code change a bit less intrusive in cases where an instrument doesn't have a goniometer.